adapter/pkg/messaging/azure_connection.go

```diff
@@ -20,22 +20,64 @@ package messaging
 
 import (
 	logger "github.com/wso2/product-microgateway/adapter/internal/loggers"
+	servicebus "github.com/Azure/azure-service-bus-go"
+	"context"
 )
 
 var (
 	// AzureRevokedTokenChannel stores the revoked token events
 	AzureRevokedTokenChannel chan []byte
+	// AzureRevokedTokenAck stores the acknowledgement for event process finish
+	AzureRevokedTokenAck chan bool
+	// AzureNotificationChannel stores the notification events
+	AzureNotificationChannel chan []byte
+	// AzureNotificationAck stores the acknowledgement for event process finish
+	AzureNotificationAck chan bool
 )
 
 func init() {
 	AzureRevokedTokenChannel = make(chan []byte)
+	AzureRevokedTokenAck = make(chan bool)
+	AzureNotificationChannel = make(chan []byte)
+	AzureNotificationAck = make(chan bool)
 }
 
-// InitiateBrokerConnection to pass event consumption
-func InitiateBrokerConnection(eventListeningEndpoint string) error {
+// InitiateBrokerConnectionAndGetAvailableTopics to initiate connection and get topic list
+func InitiateBrokerConnectionAndGetAvailableTopics(eventListeningEndpoint string) (*servicebus.Namespace,
+	[]*servicebus.TopicEntity, error) {
 	var err error
+	var availableTopics []*servicebus.TopicEntity
 	logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Trying to connect to azure service bus with " +
 		"connection string " + eventListeningEndpoint)
-	err = nil
-	return err
+	namespace, err := servicebus.NewNamespace(servicebus.NamespaceWithConnectionString(eventListeningEndpoint))
+
+	if err == nil {
+		logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Successfully received namespace ")
+		topicManager := namespace.NewTopicManager()
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		availableTopics, err = topicManager.List(ctx)
+		if err != nil {
+			logger.LoggerMgw.Errorf("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error occurred while trying to get topic " +
+				"list from azure service bus :%v", err)
+		}
+		logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Successfully received topic list ")
+	} else {
```

Comment:
We should retry these cases

Reply:
Retry will be added next .. will add todo's